packages/astro/src/content/vite-plugin-content-imports.ts

```diff
@@ -41,116 +45,201 @@ export function astroContentImportPlugin({
+			async load(viteId) {
+				if (!contentEntryExts.some((ext) => viteId.endsWith(ext))) return;
+
+				const { fileId } = getFileInfo(viteId, settings.config);
+				for (const contentEntryType of settings.contentEntryTypes) {
+					if (contentEntryType.getRenderModule) {
+						const entry = await getContentEntryModuleFromCache(fileId);
+						// Cached entry must exist (or be in-flight) when importing the module via content collections.
+						// This is ensured by the `astro:content-imports` plugin.
+						if (!entry)
+							throw new AstroError({
+								...AstroErrorData.UnknownContentCollectionError,
+								message: `Unable to render ${JSON.stringify(
+									fileId
+								)}. Did you import this module directly without using a content collection query?`,
+							});
+
+						return contentEntryType.getRenderModule({ entry });
+					}
+				}
+			},
+		});
+	}
+
+	// Used by the `render-module` plugin to avoid double-parsing your schema
+	const contentEntryModuleByIdCache = new Map<string, ContentEntryModule | 'loading'>();
+	const awaitingCacheById = new Map<string, ((val: ContentEntryModule) => void)[]>();
+	function getContentEntryModuleFromCache(id: string) {
+		const value = contentEntryModuleByIdCache.get(id);
+		// It's possible for Vite to load modules that depend on this cache
+		// before the cache is populated. In that case, we queue a promise
+		// to be resolved by `setContentEntryModuleCache`.
+		if (value === 'loading') {
+			return new Promise<ContentEntryModule>((resolve, reject) => {
+				const awaiting = awaitingCacheById.get(id) ?? [];
+				awaiting.push(resolve);
+				awaitingCacheById.set(id, awaiting);
+			});
+		} else if (value) {
+			return Promise.resolve(value);
+		}
+		return Promise.resolve(undefined);
+	}
+
+	async function setContentEntryModuleCache({
+		fileId,
+		pluginContext,
+	}: {
+		fileId: string;
+		pluginContext: PluginContext;
+	}): Promise<ContentEntryModule> {
+		contentEntryModuleByIdCache.set(fileId, 'loading');
+		const observable = globalContentConfigObserver.get();
+
+		// Content config should be loaded before this plugin is used
+		if (observable.status === 'init') {
+			throw new AstroError({
+				...AstroErrorData.UnknownContentCollectionError,
+				message: 'Content config failed to load.',
+			});
+		}
+		if (observable.status === 'error') {
+			// Throw here to bubble content config errors
+			// to the error overlay in development
+			throw observable.error;
+		}
+
+		let contentConfig: ContentConfig | undefined =
+			observable.status === 'loaded' ? observable.config : undefined;
+		if (observable.status === 'loading') {
+			// Wait for config to load
+			contentConfig = await new Promise((resolve) => {
+				const unsubscribe = globalContentConfigObserver.subscribe((ctx) => {
+					if (ctx.status === 'loaded') {
+						resolve(ctx.config);
+						unsubscribe();
+					} else if (ctx.status === 'error') {
+						resolve(undefined);
+						unsubscribe();
+					}
```

Comment:
Here we have an `if` and `else if`, which means that the two conditions might hit `false`. Is it intentional?

Reply:
Yes, this is intentional! Just means we'll keep waiting until one of these two conditions are met. We can change the `else if` to just an `if`